models/evc.py

```diff
@@ -1056,35 +1090,43 @@ def _prepare_push_flow(self, *args, queue_id=None):
         """
         # assign all arguments
         in_interface, out_interface, in_vlan, out_vlan, new_c_vlan = args
-        is_EVPL = (in_vlan is not None)
         flow_mod = self._prepare_flow_mod(
-            in_interface, out_interface, queue_id, is_EVPL
+            in_interface, out_interface, queue_id, in_vlan
         )
-
         # the service tag must be always pushed
         new_action = {"action_type": "set_vlan", "vlan_id": out_vlan}
         flow_mod["actions"].insert(0, new_action)
 
         new_action = {"action_type": "push_vlan", "tag_type": "s"}
         flow_mod["actions"].insert(0, new_action)
 
-        if in_vlan:
+        if in_vlan is not None:
             # if in_vlan is set, it must be included in the match
             flow_mod["match"]["dl_vlan"] = in_vlan
-        if new_c_vlan:
-            # new_in_vlan is set, so an action to set it is necessary
+
+        if new_c_vlan not in {"4096/4096", 0, None}:
+            # new_in_vlan is an integer but zero, action to set is required
             new_action = {"action_type": "set_vlan", "vlan_id": new_c_vlan}
             flow_mod["actions"].insert(0, new_action)
-            if not in_vlan:
-                # new_in_vlan is set, but in_vlan is not, so there was no
-                # vlan set; then it is set now
-                new_action = {"action_type": "push_vlan", "tag_type": "c"}
-                flow_mod["actions"].insert(0, new_action)
-        elif in_vlan:
-            # in_vlan is set, but new_in_vlan is not, so the existing vlan
-            # must be removed
+
+        if in_vlan not in {"4096/4096", 0, None} and new_c_vlan == 0:
```

Comment:
```suggestion
        if in_vlan not in self.special_cases and new_c_vlan == 0:
```

Reply:
I applied all of these suggestions in the last commit [5c98c93c79bffd79edb81fb824d46250e4db683e](https://github.com/kytos-ng/mef_eline/pull/258/commits/5c98c93c79bffd79edb81fb824d46250e4db683e)